File: app/src/main/java/com/koko/www/androiddemo/sqlite/SqliteActivity.java
package com.koko.www.androiddemo.sqlite;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.ItemTouchHelper;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import com.koko.www.androiddemo.R;


public class SqliteActivity extends AppCompatActivity {
    private SQLiteDatabase database;
    private GuestListAdapter guestListAdapter;
    private EditText newGuestNameEditText, newPartySizeEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sqllite);
        RecyclerView waitlistRecyclerView;
        newGuestNameEditText = findViewById(R.id.person_name_edit_text);
        newPartySizeEditText = findViewById(R.id.party_count_edit_text);
        waitlistRecyclerView = this.findViewById(R.id.all_guests_list_view);
        waitlistRecyclerView.setLayoutManager(new LinearLayoutManager(this));

        // Create a DB helper (this will create the DB if run for the first time)
        WaitlistDbHelper waitlistDbHelper = new WaitlistDbHelper(this);

        // Keep a reference to the mDb until paused or killed. Get a writable database
        // because you will be adding restaurant customers
        database = waitlistDbHelper.getWritableDatabase();

        //Fill the database with fake data
        TestUtil.insertFakeData(database);
        // Create an adapter for that cursor to display the data
        Cursor cursor = getAllGuests();


        guestListAdapter = new GuestListAdapter(this, cursor);
        waitlistRecyclerView.setAdapter(guestListAdapter);

        /**
         * 为recycleView添加滑动删除
         */
        new ItemTouchHelper(new ItemTouchHelper.SimpleCallback(0,
                ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {
            @Override
            public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
                return false;
            }

            @Override
            public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
                long id = (long) viewHolder.itemView.getTag();
                removeGuest(id);
                guestListAdapter.swapCursor(getAllGuests());
            }
        }).attachToRecyclerView(waitlistRecyclerView);

    }

    private Cursor getAllGuests() {
        return database.query(
                WaitlistContract.WaitlistEntry.TABLE_NAME,
                null,
                null,
                null,
                null,
                null,
                WaitlistContract.WaitlistEntry.COLUMN_TIMESTAMP
        );
    }

    public void addToWaitlist(View view) {
        if (newGuestNameEditText.getText().length() == 0 ||
                newPartySizeEditText.getText().length() == 0) {
            return;
        }
        int partySize = 1;
        try {

            partySize = Integer.parseInt(newPartySizeEditText.getText().toString());
        } catch (Exception e) {
            Log.e("TAG", e.getMessage());
        }
        addNewGuest(newGuestNameEditText.getText().toString(), partySize);

        guestListAdapter.swapCursor(getAllGuests());

        newPartySizeEditText.clearFocus();
        newGuestNameEditText.getText().clear();
        newPartySizeEditText.getText().clear();
    }

    private long addNewGuest(String name, int partySize) {
        ContentValues cv = new ContentValues();
        cv.put(WaitlistContract.WaitlistEntry.COLUMN_GUEST_NAME, name);
        cv.put(WaitlistContract.WaitlistEntry.COLUMN_PARTY_SIZE, partySize);

        return database.insert(WaitlistContract.WaitlistEntry.TABLE_NAME, null, cv);
    }

    private boolean removeGuest(long id) {
        return database.delete(WaitlistContract.WaitlistEntry.TABLE_NAME,
                WaitlistContract.WaitlistEntry._ID + "=" + id, null) > 0;

    }
}

File: app/src/main/java/com/koko/www/androiddemo/sqlite/SqliteActivity.kt
package com.koko.www.androiddemo.sqlite

import android.content.ContentValues
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.ItemTouchHelper
import android.widget.EditText

import com.koko.www.androiddemo.R
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.COLUMN_GUEST_NAME
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.COLUMN_PARTY_SIZE
import com.koko.www.androiddemo.utils.TestUtil
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.COLUMN_TIMESTAMP
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.ID
import com.koko.www.androiddemo.sqlite.WaitListContract.WaitListEntry.Companion.TABLE_NAME

/**
 * 目前用Room组件来操作数据库了
 */
class SqliteActivity : AppCompatActivity() {
    private var database: SQLiteDatabase? = null
    private var guestListAdapter: GuestListAdapter? = null
    private var newGuestNameEditText: EditText? = null
    private var newPartySizeEditText: EditText? = null

    private val allGuests: Cursor
        get() = database!!.query(
                TABLE_NAME, null, null, null, null, null,
                COLUMN_TIMESTAMP
        )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sqllite)
        val waitListRecyclerView: RecyclerView = this.findViewById(R.id.all_guests_list_view)
        newGuestNameEditText = findViewById(R.id.person_name_edit_text)
        newPartySizeEditText = findViewById(R.id.party_count_edit_text)
        waitListRecyclerView.layoutManager = LinearLayoutManager(this)

        // Create a DB helper (this will create the DB if run for the first time)
        val waitListDbHelper = WaitListDbHelper(this)

        // Keep a reference to the mDb until paused or killed. Get a writable database
        // because you will be adding restaurant customers
        database = waitListDbHelper.writableDatabase


        //Fill the database with fake data
        TestUtil.insertFakeData(database)
        // Create an adapter for that cursor to display the data
        val cursor = allGuests


        guestListAdapter = GuestListAdapter(this, cursor)
        waitListRecyclerView.adapter = guestListAdapter

        /**
         * 为recycleView添加滑动删除
         */
        ItemTouchHelper(object : ItemTouchHelper.SimpleCallback(0,
                ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
            override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
                val id = viewHolder.itemView.tag as Long
                removeGuest(id)
                guestListAdapter!!.swapCursor(allGuests)
            }
        }).attachToRecyclerView(waitListRecyclerView)

    }

    fun addToWaitList(view: View) {
        if (newGuestNameEditText!!.text.isEmpty() || newPartySizeEditText!!.text.isEmpty()) {
            return
        }
        Log.e("koko", "插入的数据都不为空")
        val partySize = Integer.parseInt(newPartySizeEditText!!.text.toString())

        addNewGuest(newGuestNameEditText!!.text.toString(), partySize)

        guestListAdapter!!.swapCursor(allGuests)

        newPartySizeEditText!!.clearFocus()
        newGuestNameEditText!!.text.clear()
        newPartySizeEditText!!.text.clear()
    }

    private fun addNewGuest(name: String, partySize: Int): Long {
        val cv = ContentValues()
        cv.put(COLUMN_GUEST_NAME, name)
        cv.put(COLUMN_PARTY_SIZE, partySize)

        return database!!.insert(TABLE_NAME, null, cv)
    }

    private fun removeGuest(id: Long): Boolean {
        return database!!.delete(TABLE_NAME,
                "$ID=$id", null) > 0
    }
}
